nexus/client.go

```diff
@@ -1,349 +1,113 @@
 package nexus
 
 import (
-	"bytes"
 	"context"
 	"encoding/json"
 	"errors"
-	"fmt"
-	"io"
-	"maps"
 	"math"
 	"net/http"
-	"net/url"
-	"strconv"
-	"strings"
 	"time"
-
-	"github.com/google/uuid"
 )
 
-// HTTPClientOptions are options for creating an [HTTPClient].
-//
-// NOTE: Experimental
-type HTTPClientOptions struct {
-	// Base URL for all requests. Required.
-	BaseURL string
-	// Service name. Required.
-	Service string
-	// A function for making HTTP requests.
-	// Defaults to [http.DefaultClient.Do].
-	HTTPCaller func(*http.Request) (*http.Response, error)
-	// A [Serializer] to customize client serialization behavior.
-	// By default the client handles JSONables, byte slices, and nil.
-	Serializer Serializer
-	// A [FailureConverter] to convert a [Failure] instance to and from an [error]. Defaults to
-	// [DefaultFailureConverter].
-	FailureConverter FailureConverter
-	// UseOperationID instructs the client to use an older format of the protocol where operation ID is sent
-	// as part of the URL path.
-	// This flag will be removed in a future release.
-	//
-	// NOTE: Experimental
-	UseOperationID bool
-}
-
-// User-Agent header set on HTTP requests.
-const userAgent = "Nexus-go-sdk/" + version
-
-const headerUserAgent = "User-Agent"
-
 var errEmptyOperationName = errors.New("empty operation name")
 
 var errEmptyOperationToken = errors.New("empty operation token")
 
-var errOperationWaitTimeout = errors.New("operation wait timeout")
-
-// Error that indicates a client encountered something unexpected in the server's response.
-type UnexpectedResponseError struct {
-	// Error message.
-	Message string
-	// Optional failure that may have been emedded in the response.
-	Failure *Failure
-	// Additional transport specific details.
-	// For HTTP, this would include the HTTP response. The response body will have already been read into memory and
-	// does not need to be closed.
-	Details any
-}
-
-// Error implements the error interface.
-func (e *UnexpectedResponseError) Error() string {
-	return e.Message
-}
-
-func newUnexpectedResponseError(message string, response *http.Response, body []byte) error {
-	var failure *Failure
-	if isMediaTypeJSON(response.Header.Get("Content-Type")) {
-		if err := json.Unmarshal(body, &failure); err == nil && failure.Message != "" {
-			message += ": " + failure.Message
-		}
-	}
-
-	return &UnexpectedResponseError{
-		Message: message,
-		Details: response,
-		Failure: failure,
-	}
-}
-
-// An HTTPClient makes Nexus service requests as defined in the [Nexus HTTP API].
+// A Client makes Nexus service requests as defined in the [Nexus API].
```

Comment:
I would remove the link to the API spec here and only put it on the HTTP transport.